homeassistant/components/rki_covid/sensor.py

```diff
@@ -0,0 +1,207 @@
+        vol.Required(CONF_DISTRICTS): vol.All(cv.ensure_list, [DISTRICT_SCHEMA]),
+    }
+)
+
+SENSORS = {
+    "count": "mdi:virus",
+    "deaths": "mdi:cross",
+    "recovered": "mdi:bottle-tonic-plus-outline",
+    "weekIncidence": "mdi:clipboard-pulse",
+    "casesPer100k": "mdi:home-group",
+    "newCases": "mdi:shield-bug",
+    "newDeaths": "mdi:shield-cross",
+    "newRecovered": "mdi:shield-sync",
+    "hospitalizationCasesBaby": "mdi:baby-face-outline",
+    "hospitalizationIncidenceBaby": "mdi:baby-face",
+    "hospitalizationCasesChildren": "mdi:account-child-outline",
+    "hospitalizationIncidenceChildren": "mdi:account-child",
+    "hospitalizationCasesTeen": "mdi:face-woman",
+    "hospitalizationIncidenceTeen": "mdi:face-woman-outline",
+    "hospitalizationCasesGrown": "mdi:face-man",
+    "hospitalizationIncidenceGrown": "mdi:face-man-outline",
+    "hospitalizationCasesSenior": "mdi:account-cowboy-hat-outline",
+    "hospitalizationIncidenceSenior": "mdi:account-cowboy-hat",
+    "hospitalizationCasesOld": "mdi:human-white-cane",
+    "hospitalizationIncidenceOld": "mdi:human-cane",
+}
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    async_add_entities: Callable,
+    discovery_info: Optional[DiscoveryInfoType] = None,
+) -> None:
+    """Set up the sensor platform."""
+    _LOGGER.debug("setup sensor for platform")
+    session = async_get_clientsession(hass)
+
+    if CONF_BASEURL in config:
+        _LOGGER.warning("Baseurl is not supported anymore.")
+
+    parser = RkiCovidParser(session)
+    coordinator = await get_coordinator(hass, parser)
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
+
+    districts = config[CONF_DISTRICTS]
+
+    sensors = [
+        RKICovidNumbersSensor(coordinator, district[CONF_DISTRICT_NAME], info_type)
+        for info_type in SENSORS
+        for district in districts
+    ]
+    async_add_entities(sensors, update_before_add=True)
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities,
+):
+    """Create sensors from a config entry in the integrations UI."""
+    _LOGGER.debug(f"create sensor from config entry {config_entry.data}")
+    session = async_get_clientsession(hass)
+    parser = RkiCovidParser(session)
+    coordinator = await get_coordinator(hass, parser)
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
+
+    try:
+        district = config_entry.data[ATTR_COUNTY]
+        sensors = [
+            RKICovidNumbersSensor(coordinator, district, info_type)
+            for info_type in SENSORS
+        ]
+        async_add_entities(sensors, update_before_add=True)
+    except KeyError:
+        _LOGGER.error("Could not determine %s from config!", ATTR_COUNTY)
```

Comment:
Why is a try needed here

Reply:
If an invalid/old config layout is used the integration setup will fail. (Integration was previous available via HACS)

Reply:
Why not check the district then compared to some list and do things depending on outcome. This try block seems overcomplicated for the job.

Reply:
done